app/components/utils/kind-selector.js

```diff
@@ -1,29 +1,32 @@
-/* eslint-disable ember/no-arrow-function-computed-properties */
-// TODO: octane-refactor
-// eslint-disable-next-line ember/no-classic-components
-import Component from '@ember/component';
-import CONFIG from 'frontend-kaleidos/utils/config';
-import EmberObject, { computed } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
 
-// TODO: octane-refactor
-// eslint-disable-next-line ember/no-classic-classes, ember/require-tagless-components
-export default Component.extend({
-  classNames: ['auk-u-mb-2'],
-  isLoading: null,
-  hideLabel: null,
+/**
+ * @argument {boolean} hideLabel Whether to hide the label of the dropdown
+ * @argument {boolean} initializeEmptyKind Whether to initialize the dropdown to the default meeting kind, "Ministerraad", if no meeting kind was provided
+ * @argument {function} setAction Action to perform once a meeting kind has been selected
+ * @argument {MeetingKindModel} kind (Optional) The meeting kind to set the dropdown to
+ */
+export default class UtilsKindSelector extends Component {
+  @service store;
+  @tracked kind = null;
+  @tracked options = this.store.peekAll('meeting-kind').sortBy('position'); // Meeting kinds get loaded in the agendas route, so we can just peek them here.
+  @tracked defaultKind = this.options.firstObject;
```

Comment:
Het lijkt mij niet de verantwoordelijkheid van deze selector-component om te bepalen wat de default is, want die is mogelijk niet op alle plaatsen in de app hetzelfde (als deze component in de toekomst op nog andere plaatsen gebruikt zou gaan worden). Het is de consumer van de component (`NewMeetingModal` en `EditMeetingModal`) die moet bepalen welke value geselecteerd is.

Een component interface als `Utils::KindSelector @selectedKind={{this.kind}} @onChange={{...}}` lijkt mij logischer. Het is de verantwoordelijkheid van de consumer om te zorgen dat `this.kind` de juiste waarde bevat (en dus de gewenste default bij initializatie).

Er moet dan ook geen interne `@tracked kind` poperty meer zijn in de `KindSelector`-component. De `@selectedKind` kan doorgegeven worden aan de `PowerSelect` en bij wijziging moet enkel de `@onChange`-handler aangeroepen worden. De `KindSelector`-component wordt dus in feite niets meer dan een wrapper rond `PowerSelect` die zorgt dat de dropdown met de juiste opties geïnitalizeerd wordt.